Bug fix: Update dependencies, remove Yarn

nadimkobeissi opened the pull request:
# Description

The Yarn dependencies used by many separate codebases in the monorepo were unmaintained, leading to a conflation of outdated dependencies, nonfunctional dependencies, dependencies that were impossible to install on macOS or on certain architectures, dependencies with security issues, deprecated dependencies...

This effort finalizes an upgrade to Yarn 3, all the while upgrading relevant dependencies to current, maintained versions, and also clearing out unused and duplicated dependencies across the board.

This doesn't *seem* to have broken anything, but I worry it still might have, so please review!

# Checklist:

nadimkobeissi committed:
Update .gitignore

nadimkobeissi committed:
Migrate to Yarn 3

nadimkobeissi committed:
Finish Yarn 3 migration

nadimkobeissi committed:
Update .gitignore

nadimkobeissi committed:
Get rid of outdated, deprecated, insecure Yarn dependencies

nadimkobeissi committed:
Revert package.json

nadimkobeissi committed:
Update GitHub workflows for Yarn 3

nadimkobeissi committed:
Add .gitignore

nadimkobeissi committed:
Fix Linting errors

nadimkobeissi commented:
Alright, fixed the CI. There's one CI test still failing _(Contracts / contracts (beta, pull_request) (pull_request))_ but I don't think that it has anything to do with this PR, and it should be investigated independently.

nadimkobeissi committed:
Merge branch 'nymtech:develop' into develop

doums commented:
Hey, for sure updating all these outdated deps is a good thing. But I'm not sure to migrate on the new yarn version `3` is a good idea. Yes updating yarn to latest should be a good thing. But since its version `2` and its zero install design (relaying to corepack so not very `0`), it differed significantly from the standard `npm`. In addition, if I'm not wrong, there is no very pros remaining to keep using yarn nowadays (npm has native support for workspaces since).

So instead of putting some effort migrating to yarn 3 I'd recommend switching on `npm` instead. But it might require more work than upgrading to yarn `3` as yarn is in use already

nadimkobeissi committed:
Fix missing type declaration

nadimkobeissi committed:
Update types for Typescript SDK <-> WASM debug link

nadimkobeissi committed:
Migrate Storybook (WIP)

nadimkobeissi committed:
Remove yarn-error.log

nadimkobeissi committed:
Add missing asset for React example

nadimkobeissi committed:
Add .bablerc.json

nadimkobeissi committed:
Storybook migration (WIP)

nadimkobeissi commented:
I am OK with removing Yarn completely (did not mean to close pull request!)

nadimkobeissi committed:
Begin removing Yarn entirely

nadimkobeissi committed:
Fix lint

nadimkobeissi committed:
validator: Remove obsolete `prefix` property

See: https://github.com/cosmos/cosmjs/issues/1291

nadimkobeissi committed:
explorer: Downgrade some minor dependencies

nadimkobeissi committed:
nym-connect: Fix desktop build env

nadimkobeissi committed:
nym-connect: Fix mobile build env

nadimkobeissi committed:
nym-wallet: Fix build env

nadimkobeissi committed:
testnet-faucet: fix Terser unsupported syntax

nadimkobeissi committed:
Merge branch 'nymtech:develop' into develop

nadimkobeissi committed:
Add testnet-faucet's package-lock.json

nadimkobeissi committed:
Update package-lock.json

nadimkobeissi committed:
Fix TS SDK examples and make them independent

The Nym TypeScript SDK comes with two main examples:

- `plain-html`
- `react-webpack-with-theme-example`

As of time of testing on the latest `develop` branch over at
@nymtech/nym, `react-webpack-with-theme-example` was broken because the
example was expected to load some of its packages from its own
`package.json` and the rest from the repository's root `package.json`,
which caused the problems outlined here:

https://stackoverflow.com/q/72413194

While addressing this issue, I noticed that the examples were not truly
independent from the rest of the repo. While I know that this is a
monorepo with mulitple workspaces and understand how that's supposed to
work, I think that specifically in the case of example folders, we need
to ensure the ability for these folders to work fully independently from
the rest of the repository, because example folders are overwhelmingly
likely to be copied out of the repo to be expected by curious third
parties to work while fully self-contained. Furthermore, this makes
resolving the above-mentioned original issue easier.

Therefore, in this commit, I did the following for both `plain-html` and
`react-webpack-with-theme-example`:

- Ensure that their `package.json`s are fully self-sufficient and that
  no packages would need to be loaded from the repository's root
  `package.json`.
- Ensure that their `tsconfig.json` are equally self-sufficient and do
  not reference the repository's root `tsconfig.json`.
- Ensure that the webpack configuration is present in each example,
  instead of being snuck into a hidden `.webpack` folder in the
  `sdk/typescript/examples` folder (this struck me as an ugly hack
  anyhow).
- Ensure that `react-webpack-with-theme-example` has direct access to a
  small SVG graphic of the Nym logo within its own folder.

I think these changes will both render the examples less likely to break
as the rest of the monorepo evolves. Freezing dependencies in time is
not appropriate for the monorepo, but it is totally appropriate for
examples since we don't really care about migrating them to the latest
best practices/security fixes for their dependencies as much.

Also, these changes will make the example folders more friendly for
third-party engineers, since they can just copy out the example folder
from the repo and have everything they need to understand how things
work within the example folder itself, instead of needing to hunt down
other files in the sprawling monorepo.

nadimkobeissi committed:
Merge branch 'nymtech:develop' into develop

nadimkobeissi committed:
Merge branch 'nymtech:develop' into develop

nadimkobeissi committed:
Minor change to DEVELOPERS.md

nadimkobeissi committed:
Update package-lock.json

nadimkobeissi committed:
Merge branch 'nymtech:develop' into develop
